zipline/pipeline/factors/factor.py

```diff
@@ -833,6 +834,84 @@ def linear_regression(self, target, regression_length, mask=NotSpecified):
             mask=mask,
         )
 
+    def winsorize(self,
+                  limits=0.05,
+                  inclusive=(False, False),
+                  mask=NotSpecified,
+                  groupby=NotSpecified):
+        """
+        Construct a Factor returns a winsorized row for results. This is useful
```

Comment:
The bounds are not clamped by a fixed percentile, rather they are generated from the input array.